lib/Runtime/Library/DateImplementation.cpp

```diff
@@ -1339,6 +1342,18 @@ namespace Js {
                     ss = ssNil;
                     break;
                 }
+                case ssMillisecondColon:
+                {
+                    AssertMsg(isNextFieldDateNegativeVersion5 == false, "isNextFieldDateNegativeVersion5 == false");
+
+                    if (lwT >= 1000)
```

Comment:
can't we just trim to 1000 with `lwMillisecond = lwT % 1000` ?
It would effectively ignore the characters after the third digit

Reply:
The thing is v8 returns NaN when millisecond is >= 1000 and `:` is used as delimiter.

Reply:
Right, looks good then